### PR TITLE
For Discussion ONLY: Tweaks for Python 2.7 & 3.6 tests to pass.

### PR DESCRIPTION
**Pull request by dougPhilips**
This is just a draft to see if the package could be made to work for both 2.7 and 3.6

It took me a few hours to get 'python setup.py test' to pass using both Python 2.7 and Python 3.6 virtual environments.
These changes are not ready for merge, but I wanted to highlight what changes were needed to get advice on what direction(s) to go in for a serious PR. I specifically avoided using the 'six' package because it felt like too big of a hammer, however if that is the direction to go in, I am fine with reworking these changes to use it.

Basically I wanted this to be a discussion _starting_ point.
Thank you!

### CODE REVIEW

**Comment by dougPhilips** [comment]
Gentle ping to see if @gradeawarrior is interested in discussing what direction these kinds of changes can go...
